hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mob/mapreduce/MobRefReporter.java

```diff
@@ -0,0 +1,485 @@
+          context.getCounter("SIZE_PER_CELL_DIGITS", Integer.toString(Integer.toString(cellsize).length())).increment(1L);
+          size += cellsize;
+          count++;
+        }
+      }
+      context.getCounter("CELLS_PER_ROW_DIGITS", Integer.toString(Long.toString(count).length())).increment(1L);
+      context.getCounter("SIZE_PER_ROW_DIGITS", Integer.toString(Long.toString(size).length())).increment(1L);
+      context.getCounter("MOB","NUM_CELLS").increment(count);
+    }
+  }
+
+  public static class MobRefReducer extends
+      Reducer<Text, ImmutableBytesWritable, Text, Text> {
+
+    TableName table;
+    String mobRegion;
+    Path mob;
+    Path archive;
+    String seperator;
+
+    /* Results that mean things are fine */
+    final Text MOB_DIR = new Text("MOB DIR");
+    final Text HLINK_RESTORE = new Text("HLINK TO ARCHIVE FOR SAME TABLE");
+    final Text HLINK_CLONE = new Text("HLINK TO ARCHIVE FOR OTHER TABLE");
+    /* Results that mean something is incorrect */
+    final Text ARCHIVE_ERROR_BAD_LINK = new Text("ARCHIVE WITH HLINK BUT NOT FROM OUR TABLE");
+    final Text ARCHIVE_ERROR_STALE = new Text("ARCHIVE BUT NO HLINKS");
+    final Text ARCHIVE_ERROR_IO = new Text("ARCHIVE BUT IOE WHILE CHECKING HLINKS");
+    /* Results that mean data is probably already gone */
+    final Text MISSING = new Text("MISSING FILE");
+    final Text HLINK_ERROR_DANGLING = new Text("HLINK BUT POINTS TO MISSING FILE");
+    final Text MISSING_IO = new Text("MISSING FILE BUT IOE WHILE CHECKING HLINKS");
+    final Base64.Encoder base64 = Base64.getEncoder();
+
+    @Override
+    public void setup(Context context) throws IOException, InterruptedException {
+      final Configuration conf = context.getConfiguration();
+      final String tableName = conf.get(TableInputFormat.INPUT_TABLE);
+      if (null == tableName) {
+        throw new IOException("Job configuration did not include table.");
+      }
+      table = TableName.valueOf(tableName);
+      mobRegion = MobUtils.getMobRegionInfo(table).getEncodedName();
+      final String family = conf.get(TableInputFormat.SCAN_COLUMN_FAMILY);
+      if (null == family) {
+        throw new IOException("Job configuration did not include column family");
+      }
+      mob = MobUtils.getMobFamilyPath(conf, table, family);
+      LOG.info("Using active mob area '{}'", mob);
+      archive = HFileArchiveUtil.getStoreArchivePath(conf, table,
+          MobUtils.getMobRegionInfo(table).getEncodedName(), family);
+      LOG.info("Using archive mob area '{}'", archive);
+      seperator = conf.get(TextOutputFormat.SEPERATOR, "\t");
+    }
+
+    @Override
+    public void reduce(Text key, Iterable<ImmutableBytesWritable> rows, Context context)
+        throws IOException, InterruptedException {
+      final Configuration conf = context.getConfiguration();
+      final String file = key.toString();
+      // active mob area
+      if (mob.getFileSystem(conf).exists(new Path(mob, file))) {
```

Comment:
hurm. I'm not sure. `FileSystem` already includes a cache for a given URI and Configuration; we should always use the same pairs here. I don't know how efficient that cache is, but trying to avoid it without a seeing a performance issue feels like premature optimization?

I also don't have telemetry from anywhere big yet. let me go dig a bit.

Reply:
Certainly, this is not performance critical code -- and likely not a bit performance sink either way.  I figured you were closer to recalling if there was a cache or not and I kept noticing the idiom as I read.